### PR TITLE
nodenext compatiblity

### PR DESCRIPTION
**Pull request by Uzlopak**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -136,8 +136,11 @@ function plugin (fastify, options, done) {
...
 }
 
-module.exports = fp(plugin, {
+module.exports = fp(fastifyBearerAuth, {
   fastify: '4.x',
   name: '@fastify/bearer-auth'
 })
+module.exports.default = fastifyBearerAuth
+module.exports.fastifyBearerAuth = fastifyBearerAuth
+
```
Comment (Eomm) [inline index.js:145]: `internals` what?!

**Review by Eomm — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (Uzlopak) [reply index.js:145]: I think you mean the next line? Tbh. out of scope of this PR.

But maybe @climba03003 has an idea about it.

Comment (climba03003) [reply index.js:145]: We can focus on refactoring first. Then, update the types afterward.

**Review by climba03003 — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (climba03003) [reply index.js:145]: I though `internals` exist for test only. So, it calls `internals`.
https://github.com/fastify/fastify-bearer-auth/commit/a72107b859b7b448cd9fc784946f06892e7e5bab

**Pull request merged**